hal/src/subghz/mod_params.rs

```diff
@@ -712,6 +712,24 @@ impl PartialOrd for LoRaBandwidth {
     }
 }
 
+impl From<u32> for LoRaBandwidth {
+    fn from(bw: u32) -> Self {
+        match bw {
+            7_810 => LoRaBandwidth::Bw7,
+            10_420 => LoRaBandwidth::Bw10,
+            15_630 => LoRaBandwidth::Bw15,
+            20_830 => LoRaBandwidth::Bw20,
+            31_250 => LoRaBandwidth::Bw31,
+            41_670 => LoRaBandwidth::Bw41,
+            62_500 => LoRaBandwidth::Bw62,
+            125_000 => LoRaBandwidth::Bw125,
+            250_000 => LoRaBandwidth::Bw250,
+            500_000 => LoRaBandwidth::Bw500,
+            _ => panic!("invalid LoRa bandwidth"),
```

Comment:
These should implement `TryFrom` instead of `From` because they are fallible.

https://doc.rust-lang.org/core/convert/trait.From.html

> **Note: This trait must not fail**. If the conversion can fail, use `TryFrom`.

Reply:
Good catch! I remember I wasn't completely sure what to do here, because the conversion can only fail if the regional data rate parameters in `lorawan-device` are set up wrong, so it felt weird to have the caller handle that error. But this function could of course be used anywhere, so it should absolutely be `TryFrom`.